include/swift/AST/DiagnosticsSema.def

```diff
@@ -4633,24 +4635,24 @@ NOTE(objc_optional_requirement_swift_rename,none,
      "requirement %1", (bool, DeclName))
 
 ERROR(witness_non_objc,none,
-      "non-'@objc' " OBJC_DIAG_SELECT " does not satisfy requirement "
-      "of '@objc' protocol %2",
-      (unsigned, DeclName, Identifier))
+      "non-'@objc' " OBJC_DIAG_SELECT_1 " %select{cannot be used as a default implementation to satisfy|does not satisfy}0 requirement "
+      "of '@objc' protocol %3",
+      (bool, unsigned, DeclName, Identifier))
```

Comment:
Does this new diagnostic wording clarify that default implementations are not allowed? What bugs me slightly is the "non objc" bit, because in my mind, it sort of indicates that adding '@objc' will fix-it (of course, if one adds it then we emit another diagnostic to let the user know that it is not allowed). I thought of an alternative wording:

```
default implementation of {requirement_name} cannot be used to satisfy '@objc' protocol requirement {requirement_name} in '@objc' protocol {protocol_name}.
```

(notice it drops the 'non-objc' bit)

Does that sound better? If not, do you have any suggestions?

Reply:
Hey Suyash, thanks for addressing this issue!

The current wording of the fix-it make it sounds like there is something wrong with this *particular* requirement, rather than the general idea of satisfying an objc protocol requirement with a default implementation.

I would instead opt for something along the lines of:

> Default implementations cannot be used to satisfy requirements of '@objc' protocols, such as {requirement_name} in '@objc' protocol {protocol_name}.

Reply:
I agree. I was also thinking of something similar but yours seems a little more compact than mine... I’ll update it. Thanks!

Reply:
Done ✅